tap_marketo/client.py

```diff
@@ -89,7 +89,13 @@ def __init__(self, endpoint, client_id, client_secret,
         self.domain = extract_domain(endpoint)
         self.client_id = client_id
         self.client_secret = client_secret
-        self.max_daily_calls = int(max_daily_calls)
+        try:
+            self.max_daily_calls = int(max_daily_calls or MAX_DAILY_CALLS)
+        except (ValueError, TypeError):
+            # Raises TypeError for None / Null Value
+            # Raises ValueError for "" / bool value
```

Comment:
can we make this a logger error like:

```LOGGER.warning("Invalid max_daily_calls was passed: \'{}\', using default value of 40,000.".format("max_daily_calls"))```

That way it's a little clearer what might have happened to folks reading the logs

Reply:
`int(max_daily_calls or MAX_DAILY_CALLS)` expression will fallback to default value for `None`, `True`/`False` and `""`.

IMO any other exception should be raised instead of falling back to default value.

```
>>> int("" or 100)
100
>>> int(True or 100)
1
>>> int(False or 100)
100
>>> int('1000' or 100)
1000
>>> int('True' or 100)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: 'True'
```

Reply:
I don't like the idea of failing the code for a non-mandatory value but suppressing it will not handle all scenarios such as, this will only result in the user observing unexpected behaviour.
- Number with "," eg: 1,000,000 (internationalisation issues)
- Negative Number
- Decimal input
- Zero as a string i.e "0" (which will result in the limit being set to Zero).

Hence i find it best to raise an error incase of any such values instead of using a fallback mechanism.

Reply:
the UI for this tap also suggested a value with a comma for the api_limit i.e `50,000` which is bound to fail.

<img width="692" alt="image" src="https://github.com/singer-io/tap-marketo/assets/20889199/de481213-f464-4fdf-bdbc-53f855c77b40">

Reply:
i have added a few more test scenarios and handled the negative / zero limit condition

Reply:
We have precedence in other taps of logging if we're going to fall back to a default value and allowing the tap to try to proceed with the default, but not really for an API quota scenario, so I'm good either way